Add missing entry for HAVE_PTY_H in config.h.  Without it, the
generated config.h will not define HAVE_PTY_H, and <pty.h> is not
included in openpty.c.  This leads to an implicit declaration of the
openpty function, causing a C99 compatibility problem.

This should go away on its own if upstream ever does a new upstream
release, as config.h.in is usually generated by autoheader.

diff --git a/config.h.in b/config.h.in
index 2d863e29d64848bf..c2e0cc979dc0fa56 100644
--- a/config.h.in
+++ b/config.h.in
@@ -79,6 +79,9 @@
 /* Define if you have the <paths.h> header file.  */
 #undef HAVE_PATHS_H
 
+/* Define if you have the <pty.h> header file.  */
+#undef HAVE_PTY_H
+
 /* Define if you have the <readline/history.h> header file.  */
 #undef HAVE_READLINE_HISTORY_H
 
